src/stateless/slider.vue

```diff
@@ -0,0 +1,545 @@
+            throw new Error('Difference between max and min is not divisible by step.')
+
+        let isValueValid = ((this.value - this.min) % this.step).toFixed(0) === '0'
+        if (!isValueValid)
+            throw new Error('Value is not a valid step.')
+
+        if (this.min < 0 || this.min > 998)
+            throw new Error('Min must be between 0 and 998.')
+
+        if (this.max > 999 || this.max <= this.min)
+            throw new Error('Max must be between min+1 and 999.')
+
+        if (this.value < this.min || this.value > this.max)
+            throw new Error('Value must be between min and max.')
+
+        this.ticksCount = 20
+        this.labelPadding = 8
+    },
+    mounted () {
+        this.isDomReady = true
+
+        // threshold for ticks disappearing under labels
+        this.minThreshold = this.$refs.min.clientWidth + this.labelPadding
+        this.maxThreshold = this.sliderWidth - this.$refs.max.clientWidth - this.labelPadding
+    },
+    methods: {
+        startDrag (e) {
+            if (this.disabled) return
+
+            this.isChanged = true
+            this.isDragging = true
+
+            this.setPosition(e)
+            window.addEventListener('mouseup', this.stopDrag)
+            window.addEventListener('mousemove', this.setPosition)
+            this.$refs.bar.focus()
+        },
+        setPosition (e) {
+            let percent = (e.clientX - this.sliderOffset) / this.sliderWidth
+            percent = Math.min(Math.max(0, percent), 0.999999)
+            let edgeStepOffset = this.stepPercentage / 2
+
+            let value = (Math.floor((percent - edgeStepOffset) / this.stepPercentage) + 1) * this.step + this.min
+            this.$emit('input', this.roundValue(value))
+
+            e.preventDefault()
+        },
+        stopDrag () {
+            this.isDragging = false
+            window.removeEventListener('mouseup', this.stopDrag)
+            window.removeEventListener('mousemove', this.setPosition)
+        },
+        onKeyboardInput (e) {
+            if (this.disabled) return
+            this.isChanged = true
+
+            let key = e.keyCode
+
+            if ((key === 38 || key === 39) && this.index < this.stepsCount)
+                this.$emit('input', this.value + this.step)
+            else if ((key === 37 || key === 40) && this.index > 0)
+                this.$emit('input', this.value - this.step)
+
+            e.preventDefault()
+        },
+        roundValue (value) {
+            let roundingFactor = 1
+            let decimalsPart = this.step.toString().split('.')[1]
+            if (decimalsPart) {
+                roundingFactor = parseInt('1' + '0'.repeat(decimalsPart.length), 10)
+            }
+            return Math.round(value * roundingFactor) / roundingFactor
+        },
+        handleInput (value) {
+            let number = parseFloat(value)
+            this.$emit('input', isNaN(number) ? null : number)
+        },
+        isUnitActive (index) {
```

Comment:
Create a new computed property `lastActiveIndex`, which should return `Math.floor(this.percentage * (this.ticksCount + 1))`. Instead of calling this method you can then just check if `index <= this.lastActiveIndex`.